### ISSUE TITLE
'x-codegen-request-body-name' - Should it work for swagger-codegen (ver 3.x+)?

### ISSUE THREAD

**Issue by Smootium** [swagger-api/swagger-codegen#10142]
**Description**
When using the online [Swagger Editor](https://editor.swagger.io/) (not sure what version; noticed on 13 Mar 2020) to convert our Swagger 2 YAML to OAS 3 specification, all of the `requestBody` in the YAML had `x-codegen-request-body-name` added to them.
Generating code from this YAML (`aspnetcore` server and `typescript-angular` client) using [3rd generation Swagger Online Generator](https://generator3.swagger.io/index.html) resulted in any/all `Request Bodies` parameters being named `body`.
All the `x-codegen-request-body-name` values were ignored.

The only documentation I found is in a [Migration from Swagger CodeGen.md](https://github.com/OpenAPITools/openapi-generator/blob/master/docs/migration-from-swagger-codegen.md) document in your (OpenAPITools) openapi-generator repo.
This document states:
>In OpenAPI spec v3, there's no body parameter, which is replaced by Request Body Object. The parameter name for Request Body is named automatically based on the model name (e.g. User). To control how the "Request Body" parameter is named, please add the vendor extension x-codegen-request-body-name to the operation 

The "model name" is also seemingly ignored when code is generated.

### ❓ If not, then why are they being added by your online Swagger Editor, when converting to OAS 3 specification?

**Swagger-codegen version**
`3.0.18` (via https://generator3.swagger.io/index.html)

**Swagger declaration file content or url**
In the OAS3 YAML, there is a `requestBody` described using `x-codegen-request-body-name`
[RestApi-xcodegen-request-body-name.yml](https://gist.github.com/Smootium/9dbcd76a7d26591689771bcf13cb6c34)

**Options used for generation**
(note: not including the spec object, for simplicity; see subsequent note in Repro Steps)
```json
{
  "lang": "aspnetcore",
  "specURL": "string",
  "type": "SERVER",
  "codegenVersion": "V3",
  "options": {
    "additionalProperties": {
      "packageName": "Test.Lib",
      "packageVersion": "0.0.1",
      "useDateTimeOffset": true,
      "returnICollection": true,
      "aspnet-core-version": "2.2"
    }
  }
}
```

**Steps to reproduce**
❗️  For simplicity sake, I'm listing the easiest way to reproduce the issue. This is not the process we use "in real life" (but the results are the same).
1. Copy the [RestApi-xcodegen-request-body-name.yml](https://gist.github.com/Smootium/9dbcd76a7d26591689771bcf13cb6c34) contents into [the online Swagger Editor](https://editor.swagger.io/)
1. Click on `Generate Server` drop-down menu.
1. Select `aspnetcore`.
1. Open the downloaded generated server content (zip file):
   1. Open the `src\IO.Swagger\Controllers` folder
   1. Inspect the `CalcSessionsApi.cs` file.
  ℹ️  I copied the contents of the CalcSessionsApi.cs file [into a gist](https://gist.github.com/Smootium/6d01e713148a667c0afd86bb8c0e9826), for easier inspection.

   1. Inspect the `CreateCalcSession` method
   1. Notice the `FromBody` parameter is still named `body`.

**Related issues/PRs**
Possibly related?
#9158 

**Additional Info**
The `x-codegen-request-body-name` are being added by the online editor to the same level as the `requestBody` in the YAML file.  
```yaml
post:
    tags:
    - calc-sessions
...
    requestBody:
      description: CalcSession creation details
      content:
        'application/json':
          schema:
            $ref: '#/components/schemas/FileOpenInfo'
      required: true
    responses:
      201:
...
      400:
        description: Invalid input
        content: {}
   x-codegen-request-body-name: session  <---------- ORIGINAL LOCATION
```
I tried moving it to be _inside_ the `requestBody` object, but that had no effect on the generated code.
```yaml
post:
    tags:
    - calc-sessions
    summary: create a new calc session
    description: Create a new calc session
    operationId: createCalcSession
    requestBody:
      x-codegen-request-body-name: session <---------- TRIED MOVING HERE
      description: CalcSession creation details
      content:
        'application/json':
          schema:
            $ref: '#/components/schemas/FileOpenInfo'
...
```

**Comment by JeffBerger** [comment]
This is a problem for me and my organization as well

**Comment by MarlonS91** [comment]
OpenAPI Generator says, 

> If your API client is using named parameters in the function call (e.g. Perl required & optional parameters, Ruby optional parameters), you will need to add `x-codegen-request-body-name` to the spec to restore the original body parameter name.

See [OpenAPI Generator Docs](https://openapi-generator.tech/docs/swagger-codegen-migration/#body-parameter-name) for additional info

**Comment by Smootium** [comment]
Thanks for the reply @MarlonS91. Sorry if it's not clear in the question/title, but our spec doc has that defined and all are named `body` regardless, in the generated code. It's like the property or value is being ignored or missed, resulting in what I presume to be a default param name being used instead.

**Comment by Smootium** [comment]
Thanks for the reply @MarlonS91 .
Sorry if it's not clear in the question/title, but our spec doc _is_ using this attribute and everything gets named `body` regardless, in the generated code. It's almost like the value is being ignored.

**Comment by vithu30** [comment]
@HugoMario, can you please provide an estimate for the fix of this issue.

**Comment by phillipuniverse** [comment]
For what it's worth, I successfully used `x-codegen-request-body-name` in the Javascript client and it correctly changed optional request body parameter names passed in as `opts`. As @Smootium pointed out, this attribute is a sibling of `requestBody` and goes directly underneath `post`/`post`/etc.

**Comment by Smootium** [comment]
@phillipuniverse I will try this again when I regenerate our code soon (maybe it's been fixed indirectly now)

**Comment by HugoMario** [comment]
hello @Smootium 

I like this way:
```
post:
    tags:
    - calc-sessions
    summary: create a new calc session
    description: Create a new calc session
    operationId: createCalcSession
    requestBody:
      x-codegen-request-body-name: session <---------- TRIED MOVING HERE
      description: CalcSession creation details
      content:
        'application/json':
          schema:
            $ref: '#/components/schemas/FileOpenInfo'
``` 
So, going to implement it to fix this issue.

**Comment by HugoMario** [comment]
@vithu30 

A couple of hours :)

**Comment by Smootium** [comment]
Thanks @HugoMario 
 What version will this be included in? Looking forward to seeing it deployed!

**Comment by HugoMario** [comment]
@Smootium 
it will be included in version `3.0.24`
